docs/content/guides/2.alokai-essentials/1.alokai-next-js/9.udl.md

```diff
@@ -0,0 +1,517 @@
+const unifiedApiExtension = createUnifiedExtension<Context, Config>()({
+  normalizers,
+  apiMethods
+});
+
+export const integrations = {
+  commerce: {
+    location: '@vsf-enterprise/sapcc-api/server',
+    configuration: {
+      OAuth: {
+        uri: process.env.SAPCC_OAUTH_URI,
+        clientId: process.env.SAPCC_OAUTH_CLIENT_ID,
+        clientSecret: process.env.SAPCC_OAUTH_CLIENT_SECRET,
+        tokenEndpoint: process.env.SAPCC_OAUTH_TOKEN_ENDPOINT,
+        tokenRevokeEndpoint: process.env.SAPCC_OAUTH_TOKEN_REVOKE_ENDPOINT,
+        cookieOptions: {
+          'vsf-sap-token': { secure: process.env.NODE_ENV !== 'development' }
+        }
+      },
+      api: {
+        uri: process.env.SAPCC_API_URI,
+        baseSiteId: process.env.DEFAULT_BASE_SITE_ID,
+        catalogId: process.env.DEFAULT_CATALOG_ID,
+        catalogVersion: process.env.DEFAULT_CATALOG_VERSION,
+        defaultLanguage: process.env.DEFAULT_LANGUAGE,
+        defaultCurrency: process.env.DEFAULT_CURRENCY
+      }
+    },
+    extensions: (extensions: ApiClientExtension[]) => [...extensions, unifiedApiExtension]
+  }
+};
+```
+
+Great! Now we have successfully installed and configured the Unified Data Layer in our Alokai Middleware. Next, we need to configure Alokai SDK to use the Unified Data Layer.
+
+### Configuring Alokai SDK
+
+In order to configure Alokai SDK with Unified Data Layer, we need to install `@vsf-enterprise/unified-sdk` package. This package is a set of Unified SDK Extensions for Alokai.
+
+Go to `apps/storefront` directory and install the `@vsf-enterprise/unified-sdk` package by running the following command:
+
+```bash
+npm install @vsf-enterprise/unified-sdk
+```
+
+Next, we need to build unified SDK module. Open `storefront/sdk/sdk.config.ts` file and add the following code:
+
+```diff
+- import { sapccModule } from '@vsf-enterprise/sapcc-sdk';
++ import { unifiedModule } from '@vsf-enterprise/unified-sdk';
+import { CreateSdkOptions, createSdk } from "@vue-storefront/next";
+
+export const { getSdk } = createSdk(
+  options,
+  ({ buildModule, middlewareUrl, getRequestHeaders }) => ({
+-    sapcc: buildModule(sapccModule, {
++    unified: buildModule(unifiedModule, {
+-     apiUrl: middlewareUrl + "/sapcc",
++     apiUrl: middlewareUrl + "/commerce",
+      requestOptions: {
+        headers: () => getRequestHeaders() as Record<string, string>,
+      },
+    }),
+  }),
+);
+```
+
+We are still missing one last thing. `unifiedModule` expects a generic type from `unifiedApiExtension` object. This is needed for SDK to automatically generate type safe methods based on Unified `apiMethods` object.
+
+Go back to `middleware.config.ts` file and add the following line at the end of the file:
+
+```ts
+export type UnifiedApiExtension = typeof unifiedApiExtension;
+```
+
+Now, go back to `storefront/sdk/sdk.config.ts` file and add the generic type to `unifiedModule`:
+
+```diff
+export const { getSdk } = createSdk(
```

Comment:
This might be picky but this diff misses import:
`import { UnifiedApiExtension } from '../../middleware/middleware.config';`